lib/core/Observer.ts

```diff
@@ -124,6 +126,7 @@ export default class Observer {
           this.transactionsUnderProcessing.push(transactionUnderProcessing);
           // Intentionally not awaiting on downloading and processing each operation batch.
           this.processTransaction(transaction, transactionUnderProcessing);
+          await this.confirmationStore.confirm(transaction.anchorString, transaction.transactionTime);
```

Comment:
Probably should do this within `processTransaction()` method given that `processTransaction()` is called in another place.